AsyncDisplayKit/Private/ASDisplayNode+UIViewBridge.mm

```diff
@@ -282,212 +288,222 @@ - (void)setNeedsDisplay
       [rasterizedContainerNode setNeedsDisplay];
     });
   } else {
-    // If not rasterized (and therefore we certainly have a view or layer),
-    // Send the message to the view/layer first, as scheduleNodeForDisplay may call -displayIfNeeded.
-    // Wrapped / synchronous nodes created with initWithView/LayerBlock: do not need scheduleNodeForDisplay,
-    // as they don't need to display in the working range at all - since at all times onscreen, one
-    // -setNeedsDisplay to the CALayer will result in a synchronous display in the next frame.
-
-    _messageToViewOrLayer(setNeedsDisplay);
-
-    BOOL nowDisplay = ASInterfaceStateIncludesDisplay(_interfaceState);
-    // FIXME: This should not need to recursively display, so create a non-recursive variant.
-    // The semantics of setNeedsDisplay (as defined by CALayer behavior) are not recursive.
-    if (_layer && !_flags.synchronous && nowDisplay && [self __implementsDisplay]) {
-      [ASDisplayNode scheduleNodeForRecursiveDisplay:self];
+    if (self.nodeLoaded) {
+      if (ASDisplayNodeThreadIsMain()) {
+        _messageToViewOrLayer(setNeedsDisplay);
```

Comment:
@Adlai-Holler this condition actually does not seem correct to me. I believe we must call __setNeedsDisplay after messaging to view or layer.

Is this a scenario where we should be using ASDisplayNodeShouldApplyBridgedWriteToView?